sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -1860,16 +1860,19 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
   override def visitTypeConstructor(ctx: TypeConstructorContext): Literal = withOrigin(ctx) {
     val value = string(ctx.STRING)
     val valueType = ctx.identifier.getText.toUpperCase(Locale.ROOT)
+    val negativeSign = Option(ctx.negativeSign).map(_.getText).getOrElse("")
+    val isNegative = if (negativeSign == "-") true else false
```

Comment:
```suggestion
    val isNegative = ctx.negativeSign != null && ctx.negativeSign.getText == "-"
```

Reply:
and remove `val negativeSign`

Reply:
do we really need to check `ctx.negativeSign.getText`? the parser rule is `MINUS? ...`

Reply:
`negativeSign` is used in error message, but I think @cloud-fan 's point is right, we need not to check `ctx.negativeSign.getText`. I'd do some tests to verify this.